pkg/android/get-android-info.go

```diff
@@ -0,0 +1,245 @@
+package android
+
+import (
+	"archive/zip"
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"github.com/bugsnag/bugsnag-cli/pkg/proto_messages"
+	"google.golang.org/protobuf/proto"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// Android Manifest reference attriubte IDs
+// https://developer.android.com/reference/android/R.attr#versionCode
+const AndroidVersionCodeId uint32 = 16843291
+
+// https://developer.android.com/reference/android/R.attr#versionName
+const AndroidVersionNameId uint32 = 16843292
+
+type AndroidManifestData struct {
+	XMLName       xml.Name                       `xml:"manifest"`
+	ApplicationId string                         `xml:"package,attr"`
+	VersionCode   string                         `xml:"versionCode,attr"`
+	VersionName   string                         `xml:"versionName,attr"`
+	Application   AndroidManifestApplicationData `xml:"application"`
+}
+
+type AndroidManifestApplicationData struct {
+	XMLName  xml.Name                `xml:"application"`
+	MetaData AndroidManifestMetaData `xml:"meta-data"`
+}
+
+type AndroidManifestMetaData struct {
+	XMLName xml.Name `xml:"meta-data"`
+	Name    []string `xml:"name,attr"`
+	Value   []string `xml:"value,attr"`
+}
+
+func isXMLContent(buffer []byte) bool {
+	xmlHeader := []byte("<manifest xmlns")
+	return bytes.Contains(buffer, xmlHeader)
+}
+
+func isProtobufContent(path string) (bool, error) {
+	// Open the file
+	file, err := os.Open(path)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	// Read the first 8 bytes of the file
+	buffer := make([]byte, 8)
+	_, err = file.Read(buffer)
+	if err != nil {
+		return false, err
+	}
+
+	// Check for the protobuf identifier
+	protobufIdentifier := []byte{0x0A} // Example: Protobuf identifier often starts with 0x0A
+	return bytes.HasPrefix(buffer, protobufIdentifier), nil
+}
+
+// Function to extract the android manifest file from a given aab file
+func GetAndroidManifestFileFromAAB(path string) (string, error) {
```

Comment:
Can this be shared with the existing AAB functionality? (Could be a later piece of work, but long-term it doesn't feel like we should have this duplicated.)

Reply:
Yes long term it will be shared.